sdk/core/azure_core/src/http/poller.rs

```diff
@@ -1,51 +1,491 @@
 // Copyright (c) Microsoft Corporation. All rights reserved.
 // Licensed under the MIT License.
 
-//! Types and methods for Long-Running Operations (LROs).
+//! Types and methods for long-running operations (LROs).
 
-use crate::http::headers::Headers;
-use typespec_client_core::time::{Duration, OffsetDateTime};
+use crate::{
+    error::ErrorKind,
+    http::{headers::Headers, Format, Response, StatusCode},
+    sleep,
+    time::{Duration, OffsetDateTime},
+};
+use futures::{stream::unfold, Stream, StreamExt};
+use serde::Deserialize;
+use std::{
+    convert::Infallible,
+    fmt,
+    future::Future,
+    pin::Pin,
+    str::FromStr,
+    task::{Context, Poll},
+};
 
-/// Default retry time for long running operations if no retry-after header is present
+/// Default retry time for long-running operations if no retry-after header is present
 ///
-/// This value is the same as the default used in the Azure SDK for Python.
-/// Ref: <https://github.com/Azure/azure-sdk-for-python/blob/main/sdk/core/azure-mgmt-core/azure/mgmt/core/polling/arm_polling.py#L191>
+/// This value is the same as the default used in other Azure SDKs e.g.,
+/// <https://github.com/Azure/azure-sdk-for-python/blob/azure-mgmt-core_1.5.0/sdk/core/azure-mgmt-core/azure/mgmt/core/polling/arm_polling.py#L191>
```

Comment:
Does it matter that this is using a constant from a management plane SDK? Do the constants differ between management and data plane (the LRO pattern is different, so I wasn't sure if the durations would also be different).